### PR TITLE
Some simple changes

### PR DESCRIPTION
**Pull request by TheAlienKnight**
Some simple changes, mostly in prep for some other fixes/additions discussed in the community discord

- Changes docker files, allowing for a less confusing setup process
- Bumps node to v18 since v17 is EOL
- Bumps Discord.js to v14, since v13 is soon to be obsolete
- Changes a few small things to allow discord.js v14 to work properly

These changes have been tested and work properly on Ubuntu 22.04.2 LTS

### CODE REVIEW

File: .docker-compose.yml
Comment (advaith1) [inline .docker-compose.yml:1]: is this file necessary? cc @IceeMC

File: src/Util/Services/discord.ts [hunk 1]
```diff
@@ -55,8 +55,7 @@ class Client extends Discord.Client {
 
 export const bot = new Client({
     allowedMentions: { parse: [] },
-    intents: ["GUILDS", "GUILD_MEMBERS", "GUILD_PRESENCES"],
-    http: { version: 8 }
+    intents: ["Guilds", "GuildMembers", "GuildPresences"],
```
Comment (advaith1) [inline src/Util/Services/discord.ts:58]: this should use the enum

**Review by advaith1 — CHANGES_REQUESTED** [review]
DEL relies extensively on `client.api` which was replaced by `client.rest` in d.js v14, so I don't see how it's possible that this code works?

**Comment by TheAlienKnight** [comment]
You are correct, it seems my docker build had not updated (I guess I forgot to prune it) and it does in fact, not operate on v14 as expected. I'll make the corrected changes tonight! Though it would sure help if discord.js documented the new discord.js/rest system, since it's entirely empty :^)

**Comment by TheAlienKnight** [comment]
Added correct changes for discord.js v14! All of these *have* actually been tested correctly this time (lol) - my bad about earlier. The changes should be broken down easily with the commit messages, and side-by-side comparisons!

File: src/Util/Services/discord.ts (hunk 1 above)
Comment (TheAlienKnight) [reply src/Util/Services/discord.ts]: Changed

**Review by advaith1 — APPROVED** [review]

**Comment by carolinaisslaying** [comment]
Is this on pause or is it safe to merge this and work on the remaining patches at a later date? Also, what else needs to be done? Would like to work on some stuff myself if possible so I'm at least somewhat productive in my spare time 😅

**Comment by TheAlienKnight** [comment]
Last I checked these fixes worked, though I had a server raid card go bad not too long ago and it screwed up the VM I had setup for testing this (didn't have a backup for it), so I can't 100% verify it works with the latest API changes. From what I can see though it should work fine!

File: views/templates/search.ejs [hunk 2]
```diff
@@ -94,7 +94,6 @@
 
         const input = document.querySelector("input.input.is-rounded.is-dark#searchInput");
 
-        if (input.value.match(/gay(\s+)?(men\s+)?.*/i)) return window.location.href = "/users/302604426781261824";
```
Comment (carolinaisslaying) [inline views/templates/search.ejs:97]: lmfao I forgot this code still existed

**Review by carolinaisslaying — APPROVED** [review]
Did a reasonable amount of testing, changes in the pull request itself work fine. Functions okay!

**Pull request merged**